File: lib/layer2_core/dark/e_inverse_primary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EInversePrimaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd24;
      case 'standard': return cstd24;
      case 'blue': return cbd24;
      case 'green': return cgd24;
      case 'red': return crd24;
      case 'nothing': return cnod24;
      default: return cstd24;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EInversePrimaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EInversePrimaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd24
            AppTheme.STANDARD -> R.color.cstd24
            AppTheme.BLUE -> R.color.cbd24
            AppTheme.GREEN -> R.color.cgd24
            AppTheme.RED -> R.color.crd24
            AppTheme.NOTHING -> R.color.cnod24
        }
    }
}
